src/thermostate/thermostate.py

```diff
@@ -395,5 +438,13 @@ def _set_properties(
                 value = self._abstract_state.keyed_output(p) * units(
                     self._SI_units[prop]
                 )
-
+                
+            if self._units == "SI":
+                set_units = getattr(default_SI, prop, None) 
+                if set_units is not None:
+                    value.ito(set_units)
```

Comment:
Since this code is that same for both branches you can simplify a bit:

```
set_units = None
if self.units == "SI":
    set_units = getattr(default_SI, prop, None)
elif self.units == "EE":
    set_units = getattr(default_EE, prop, None)
if set_units is not None:
    value.ito(set_units)
setattr(self, "_" + prop, value)
```